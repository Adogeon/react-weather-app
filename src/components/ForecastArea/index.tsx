import { HourlyForecast } from "../../types/weather";
import { parsingTimeData } from "../../utils/parsingTimeData";
import HourlyForcastCard from "../HourlyForecastCard";

export interface ForecastAreaProps {
  currentTime: string;
  hour: Array<HourlyForecast>;
}

const ForecastArea = ({ data }: { data: ForecastAreaProps }) => {
  const { dayString: currentDay, hourString: currentHour } = parsingTimeData(
    data.currentTime
  );
  let displayHour = data.hour;
  if (data.hour[0].time.match(currentDay)) {
    const currentHourIndex = displayHour.findIndex((hour) => {
      const { hourString } = parsingTimeData(hour.time);
      return hourString.match(currentHour);
    });
    displayHour = displayHour.slice(currentHourIndex + 1);
  }

  return (
    <section id="forecast">
      {displayHour.map((hour) => (
        <HourlyForcastCard data={hour} key={hour.time} />
      ))}
    </section>
  );
};

export default ForecastArea;
