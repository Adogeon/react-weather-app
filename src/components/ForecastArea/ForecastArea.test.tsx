import { render } from "@testing-library/react";
import { test, expect } from "vitest";
import ForecastArea, { type ForecastAreaProps } from ".";

const TestData: ForecastAreaProps = {
  currentTime: "test-date 14:32",
  hour: [
    {
      time: "test-date 13:00",
      temp_c: "17",
      temp_f: "62.6",
      feelslike_c: "17",
      feelslike_f: "62.6",
      condition: {
        text: "test-condition",
        icon: "/64x64/test-iocn",
      },
    },
    {
      time: "test-date 14:00",
      temp_c: "17",
      temp_f: "62.6",
      feelslike_c: "17",
      feelslike_f: "62.6",
      condition: {
        text: "test-condition",
        icon: "/64x64/test-iocn",
      },
    },
    {
      time: "test-date 15:00",
      temp_c: "17",
      temp_f: "62.6",
      feelslike_c: "17",
      feelslike_f: "62.6",
      condition: {
        text: "test-condition",
        icon: "/64x64/test-iocn",
      },
    },
  ],
};

test("ForecastArea should render correctly", () => {
  const { asFragment } = render(<ForecastArea data={TestData} />);

  expect(asFragment()).toMatchSnapshot();
});
