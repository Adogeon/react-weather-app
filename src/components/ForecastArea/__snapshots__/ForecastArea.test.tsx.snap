// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ForecastArea should render correctly 1`] = `
<DocumentFragment>
  <section
    id="forecast"
  >
    <div
      class="_hourly-card_0df995"
    >
      <div>
        14 PM
      </div>
      <div>
        <img
          alt="test-condition-icon"
          src="/64x64/test-iocn"
        />
      </div>
      <div>
        17°C
      </div>
    </div>
    <div
      class="_hourly-card_0df995"
    >
      <div>
        15 PM
      </div>
      <div>
        <img
          alt="test-condition-icon"
          src="/64x64/test-iocn"
        />
      </div>
      <div>
        17°C
      </div>
    </div>
  </section>
</DocumentFragment>
`;
